File: java-suite/src/main/java/de/fraunhofer/iem/basics/equality/withoutEqualsAndHashCode/Person.java
package de.fraunhofer.iem.basics.equality.withoutEqualsAndHashCode;

/**
 * Helper class for equality example
 *
 * @author Ranjith Krishnamurthy
 */
public class Person {
    private String name;
    private String address;
    private int age;
    private String mobileNumber;

    public Person(String name, String address, int age, String mobileNumber) {
        this.name = name;
        this.address = address;
        this.age = age;
        this.mobileNumber = mobileNumber;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getMobileNumber() {
        return mobileNumber;
    }

    public void setMobileNumber(String mobileNumber) {
        this.mobileNumber = mobileNumber;
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/uniqueToKotlin/destructuringDeclaration/inCustomClass/Person.kt
package de.fraunhofer.iem.uniqueToKotlin.destructuringDeclaration.inCustomClass

/**
 * Helper class to demonstrate the destructuring declaration.
 *
 * Note:
 * 1. Normal class by default does not contain a componentN function. Therefore, we need to explicitly define componentN
 * function as member or extension function with the operator modifier.
 * 2. componentN function can return any order value depending on the implementation
 * for example, here component1 function instead of returning name it is returning mobileNumber
 *
 * @author Ranjith Krishnamurthy
 */
class Person(
    val name: String,
    val address: String,
    val age: Int,
    val mobileNumber: String
) {
    operator fun component1(): String {
        return this.mobileNumber
    }

    operator fun component3(): String {
        return this.address
    }
}